src/Emulator/DialogueBox/DialogueBox.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { useRef, useEffect } from 'react';
+import { Icon } from '@chakra-ui/core';
+import { useDialogue } from '../EmulatorProvider';
+
+import { DialogueBoxContainer, InnerContainer, StyledClearbutton } from './styled';
+
+const DialogueBox = () => {
+  const { dialogueMessages, clearDialogue } = useDialogue();
+
+  useEffect(() => {
+    if (containerScrollRef.current){
+      // Convert to HTMLDivElement, because TypeScript complains that
+      // .scrollTop and .scrollHeight does not exist.
+      // Convert to unknown first, because TypeScript complains that
+      // 'null' and 'HTMLDivElement' do not sufficiently overlap.
+      const element = containerScrollRef.current as unknown as HTMLDivElement;
+      element.scrollTop = element.scrollHeight;
+    }
+  }, [dialogueMessages]);
+
+  const containerScrollRef = useRef(null);
```

Comment:
You can also use a generic type of `HTMLDivElement` with the `useRef` instead of using `unknown`.

```suggestion
      const element = containerScrollRef.current;
      element.scrollTop = element.scrollHeight;
    }
  }, [dialogueMessages]);

  const containerScrollRef = useRef< HTMLDivElement>(null);
```